hi-ml-histopathology/src/histopathology/configs/classification/DeepSMILECrck.py

```diff
@@ -108,12 +108,16 @@ def setup(self) -> None:
 
     def get_data_module(self) -> TilesDataModule:
         image_key = TcgaCrck_TilesDataset.IMAGE_COLUMN
-        transform = Compose(
-            [
+        if self.is_finetune:
+            transform = LoadTilesBatchd(image_key, progress=True)
+            dataloader_kwargs = dict(num_workers=os.cpu_count(), pin_memory=True)
```

Comment:
here and below: `num_workers==os.cpu_count()` is quite likely too much. To the best of my knowledge, this will turn out to be number of workers per GPU. So if you have 8 GPUs, you're overloading the GPU by a factor of 8. This will increase the chance of process starvation. `floor(cpu_count / gpu_count)`?

Reply:
Happy to try this, but I was following the guidance I found online. For example, from the Lightning docs on DDP: https://pytorch-lightning.readthedocs.io/en/latest/guides/speed.html#num-workers.

Reply:
Updated now following your suggestion and some further advice I found online.